pygame/math.py

```diff
@@ -0,0 +1,1575 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public
+# License along with this library; if not, write to the Free
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+# MA  02110-1301  USA
+
+""" Math module """
+
+from __future__ import absolute_import, division
+import math
+from numbers import Number
+
+from pygame.compat import string_types, pow_compat
+
+VECTOR_EPSILON = 1e-6  # For equality tests
+
+
+def enable_swizzling():
+    """Globally enables swizzling for vectors.
+
+    Enables swizzling for all vectors until disable_swizzling() is called.
+    By default swizzling is disabled.
+
+    Note:  There may be performance degradation, as this replaces __getattr__
+           and __setattr__.  Only enable if required.
+    """
+    # __getattr__ is not defined by default, so nothing to save
+    Vector2.__getattr__ = Vector2.__getattr_swizzle__
+    Vector3.__getattr__ = Vector3.__getattr_swizzle__
+
+    if '__oldsetattr__' not in dir(Vector2):
+        Vector2.__oldsetattr__ = Vector2.__setattr__
+    Vector2.__setattr__ = Vector2.__setattr_swizzle__
+
+    if '__oldsetattr__' not in dir(Vector3):
+        Vector3.__oldsetattr__ = Vector3.__setattr__
+    Vector3.__setattr__ = Vector3.__setattr_swizzle__
+
+
+def disable_swizzling():
+    """Globally disables swizzling for vectors.
+
+    Disables swizzling for all vectors until enable_swizzling() is called.
+    By default swizzling is disabled.
+    """
+    if '__getattr__' in dir(Vector2):
+        del Vector2.__getattr__
+    if '__oldsetattr__' in dir(Vector2):
+        Vector2.__setattr__ = Vector2.__oldsetattr__
+    if '__getattr__' in dir(Vector3):
+        del Vector3.__getattr__
+    if '__oldsetattr__' in dir(Vector3):
+        Vector3.__setattr__ = Vector3.__oldsetattr__
+
+
+class Vector2(object):
+    """A 2-Dimensional Vector."""
+
+    def __init__(self, *args):
+        if len(args) == 0:
+            self.x = 0.0
+            self.y = 0.0
+        elif len(args) == 1:
+            if isinstance(args[0], Vector2):
+                self.x = args[0].x
+                self.y = args[0].y
+            elif isinstance(args[0], string_types):
+                if args[0].startswith("<Vector2(") and args[0].endswith(")>"):
+                    tokens = args[0][9:-2].split(",")
+                    try:
+                        self.x = float(tokens[0])
+                        self.y = float(tokens[1])
+                    except ValueError:
+                        raise TypeError("Invalid string arguments - not floats ({}).".
+                                        format(args[0]))
+                else:
+                    raise TypeError("Invalid string argument - not like __repr__ ({}).".
+                                    format(args[0]))
```

Comment:
pygame supports initiliasing Vector2 with a single number:
```
>>> Vector2(2)
<Vector2(2, 0)>
```
This should either be supported, or the discrepency should be noted.